src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java

```diff
@@ -166,8 +166,11 @@ public void goToPoseWithPID(Pose2d targetPose) {
     ChassisSpeeds currentSpeed =
         ChassisSpeeds.fromRobotRelativeSpeeds(getState().Speeds, getRotation());
 
-    PID_X.reset(getPose().getX(), currentSpeed.vxMetersPerSecond * 0.4);
-    PID_Y.reset(getPose().getY(), currentSpeed.vyMetersPerSecond * 0.4);
+    double predicted_X = (targetPose.getX() - getPose().getX()) * 0.4 + getPose().getX();
```

Comment:
@isaacngkh what is this code trying to do?   It deserves a comment explaining the logic.
also consider using built in function
`double predicted_X = MathUtil.interpolate(curPose2d.getX(), targetPose.getX(), 0.4);`